File: modules/@angular/router/src/metadata/decorators.ts
import {makeDecorator} from '../core_private';

import {RouteMetadata, RoutesMetadata} from './metadata';


/**
 * Defines routes for a given component.
 *
 * It takes an array of {@link RouteMetadata}s.
 */
export interface RoutesFactory {
  (routes: RouteMetadata[]): any;
  new (routes: RouteMetadata[]): RoutesMetadata;
}

/**
 * Defines routes for a given component.
 *
 * It takes an array of {@link RouteMetadata}s.
 * @Annotation
 */
export var Routes: RoutesFactory = <RoutesFactory>makeDecorator(RoutesMetadata);

File: modules/@angular/router/src/metadata/decorators.dart
library angular.alt_router.decorators;

import 'metadata.dart';
export 'metadata.dart';

/**
 * Defines routes for a given component.
 *
 * It takes an array of {@link RouteMetadata}s.
 */
class Routes extends RoutesMetadata {
  const Routes(List<RouteMetadata> routes): super(routes);
}